[Kenneth Hong Jia Wei] iP

knotstoks opened the pull request:
# **This is Pix.** 🧚🏻 
## The Tilted Task Manager.
>How can you face your [problems](https://www.youtube.com/watch?v=Ys4WPXqkobs&ab_channel=GarenMain), if the problem is your face?

## What does Pix do??
* Type in commands!
* Get responses to those commands!
* Task list!

## Do you want to use it?
1. Go to [someone else's repo](https://github.com/nus-cs2103-AY2122S1/ip/pull/57)
2. Download it from there

## Things to note!
- [ ] Has 100% great software engineering principles 
- [x] The coder is trying his **hardest** 😢 

Here's an emoji of some *doggo tosey woseys:* 🐾 

## I am using `<inline code>` here

Commit:
Add Gradle support

knotstoks committed:
Level 0

knotstoks committed:
Level 1

knotstoks committed:
Level 2

knotstoks committed:
Level 3

knotstoks committed:
Level 4

knotstoks committed:
Level 5

knotstoks committed:
Level 6

knotstoks committed:
A-TextUiTesting and A-Enums

knotstoks committed:
Edits to A-TextUiTesting

knotstoks committed:
Edit the code to follow Java Coding conventions.

knotstoks committed:
Add the saving feature for Pix, which saves in file Pix.txt

knotstoks committed:
Add the feature that detects correct date formats in deadline and event tasks.

knotstoks committed:
Commit before branching

knotstoks committed:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Pix.java

knotstoks committed:
Merge branch 'master' into branch-Level-8

knotstoks committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Pix.java

knotstoks committed:
Fix bugs for date and formatting for date for deadlines and events.

knotstoks committed:
Refactor code into multiple classes and used more OOP concepts.

knotstoks committed:
Add JUnit tests for Todo and Event classes.

knotstoks committed:
Release version 1.0 of Pix.

knotstoks committed:
Add Java Doc to all the files in the project

knotstoks committed:
Format all project files to the Java Coding Standard.

knotstoks committed:
Add the find command to find tasks from the task list.

knotstoks committed:
Merge branch 'branch-A-JavaDoc'

knotstoks committed:
Merge branch 'branch-A-CodingStandard'

knotstoks committed:
Merge branch 'branch-Level-9'

knotstoks committed:
Merge remote-tracking branch 'origin/add-gradle-support'

knotstoks committed:
Add Gradle support to Pix.

knotstoks committed:
Add checkstyle with Gradle and fix stylechecks with the project.

knotstoks committed:
Merge branch 'branch-CheckStyle'

knotstoks committed:
Add GUI to Pix.

knotstoks committed:
Merge branch 'branch-Level-10'

wyrchris commented on src/main/java/pix/Pix.java:
```diff
@@ -0,0 +1,50 @@
+import pix.task.TaskList;
+import pix.ui.Ui;
+
+public class Pix {
```
Missing header comments for public class

wyrchris commented on src/main/java/pix/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+
+public abstract class Task {
+    protected String name;
+    protected boolean done;
```
Could rename as isDone for greater clarity

wyrchris reviewed (commented):
Other than missing header comments for public classes, or else looks good!

knotstoks committed:
Fix coding standard using comments from the PR review.

knotstoks committed:
Edit build.gradle to be able to generate a jar file.

knotstoks committed:
Add assertions to the code.

There are instances where the program should stop
running to prevent unwanted behaviours.

Adding assertions helps stop the program when these situations occur.

knotstoks committed:
Improve quality of code.

Assists other users to be able to read the code easier by improving the readibility.

Deleting old code and making the 'Happy Path' more prominent.

knotstoks committed:
Merge pull request #1 from knotstoks/branch-A-Assertions

Add assertions to the code.

knotstoks committed:
Merge pull request #2 from knotstoks/branch-A-CodeQuality

Improve quality of code.

knotstoks committed:
Add the undo feature to undo the previous command.

knotstoks committed:
Merge branch 'branch-C-Undo'

hungkhoaitay reviewed (commented):
The code is well written. I couldn't find any coding standard violations. Great job! 💯

Samuel-bit-prog commented on src/main/java/pix/Pix.java:
```diff
@@ -0,0 +1,93 @@
+     *
+     * @return Returns the last command inputted.
+     */
+    public Command getLastCommand() {
```
Maybe make lastCommand and taskList protected instead of private so do not need getTaskList and getLastCommand

Samuel-bit-prog commented on src/main/java/pix/javafx/MainWindow.java:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
```
The javadocs make it easier to understand. Very good job

Samuel-bit-prog commented on src/main/java/pix/parser/Parser.java:
```diff
@@ -0,0 +1,238 @@
+     * @return Returns the command for exiting Pix.
+     * @throws PixInvalidTaskException Throws the exception if the format is incorrect.
+     */
+    private static Command getExitCommand(String[] splitCommand) throws PixInvalidTaskException {
```
Maybe instead of getExitCommand, can use confirmExitCommand. Same with the rest.

Samuel-bit-prog commented on src/main/java/pix/task/TaskList.java:
```diff
@@ -0,0 +1,177 @@
+     * @return Returns the string representation of the task deleted.
+     */
+    public String deleteTask(int n) throws PixException {
+        String res = "";
```
Maybe instead of res, can change to reply to make it more clear

Samuel-bit-prog reviewed (commented):
Looks good. Could not really find any coding standard violations. Keep up the good work!

knotstoks committed:
Set theme jekyll-theme-tactile

knotstoks committed:
Add User Guide and Ui.png.

knotstoks committed:
Merge branch 'master' of https://github.com/knotstoks/ip

knotstoks committed:
Build Pix for release.

knotstoks committed:
Fix README

knotstoks committed:
Delete Ui.PNG

knotstoks committed:
Add files via upload
